codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -765,5 +765,28 @@ public State state() {
         public String toString() {
             return String.valueOf(id());
         }
+
+        @Override
+        public int hashCode() {
+            Http2Stream stream = this.stream;
+            if (stream == null) {
+                return id;
+            }
+            return stream.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            DefaultHttp2FrameStream that = (DefaultHttp2FrameStream) o;
+            Http2Stream stream = this.stream;
+            Http2Stream thatStream = that.stream;
+            return id == that.id && stream != null && stream.equals(thatStream);
```

Comment:
Are two `DefaultHttp2FrameStream` considered equal if `this.stream == that.stream == null`?

Reply:
I would say yes, but it's a good point @bryce-anderson (maybe I'm wrong!)

Reply:
I think they'd be considered equal as well, although I don't believe it's covered by this implementation.
This would probably do it:
```
return this.id == that.id && Objects.equals(this.stream, that.stream);
```